experimental/dispatch_profiler/profiler.py

```diff
@@ -39,55 +40,56 @@
-  for operation_kind, operation_collection_list in manifest.operations.items():
-    for operation_collection in operation_collection_list:
-
-      # Select and create an instance of operation_launcher for the operation with operation_kind.
-      # print(operation_collection.operation.name())
-      operation_launcher = operation_launcher_map[operation_kind](
-          args, operation_collection.operation)
-
-      for configuration in operation_collection.configuration_list:
-
-        # Compile the operation dispatches for verification and profiling.
-        if args.compile_only:
-          operation_launcher.compile(CompilationMode.Verify)
-          operation_launcher.compile(CompilationMode.Profile)
-
-        else:
-          # Initialize verification and profiling results.
-          verification_result = 'Not verified' if not args.verification_enabled else 'Failed'
-          runtime = -1.0
-
-          # Launch the operation dispatches for verification and profiling.
-          if args.verification_enabled:
-            verification_result = operation_launcher.verify(configuration)
-          if args.profiling_enabled:
-            runtime = operation_launcher.profile(configuration)
-
-          # Save and print the performance result.
-          if args.verification_enabled or args.profiling_enabled:
-            # Create and print a performance result.
-            result = PerformanceResult(operation_collection.operation,
-                                       configuration, verification_result,
-                                       runtime)
-            result.print()
-
-            # Append the performance result to the performance report.
-            perf_report.append_perf_result(result)
+  # For all the operations in the manifest compile (if needed), verify, and profile.
+  for _, dispatch_collection_list in manifest.dispatch_collection_map.items():
+    for dispatch_collection in dispatch_collection_list:
+
+      operation = dispatch_collection.operation
+      # Select and create an instance of operation_launcher for the operation.
+      operation_launcher = IreeToolsLauncher(args, operation)
+      for configuration in dispatch_collection.configuration_list:
+
+        # Create a dispatch object.
+        dispatch = Dispatch(operation, configuration)
+
+        # Skip the dispatch if filter returns false.
+        if not manifest.is_enabled(dispatch):
+          continue
+
+        # If dry run is enabled, skip the dispatch.
+        if args.dry_run:
+          print(f'[Dry run] : {dispatch.name()}')
+          continue
+
+        # Initialize verification and profiling results.
+        verification_result = 'Not verified' if not args.verification_enabled else 'Failed'
+        runtime = -1.0
+
+        # Launch the operation dispatches for verification and profiling.
+        if args.verification_enabled:
+          verification_result = operation_launcher.verify(configuration)
```

Comment:
To make this become a test in CI/local build (as we discussed in the email), how to disable some verifications when it is necessary to do so? (like we use xfail in compiler tests).

I'm wondering if we need to write the verification with pytest (https://docs.pytest.org/en/7.3.x/). It supports parameterized tests so we should be able to use the generated test matrix.

Reply:
We can separate out the launches where verification-enabled flag can be set to false `--verification-enabled=false`